rllib/algorithms/algorithm.py

```diff
@@ -2413,11 +2431,15 @@ def _compile_iteration_results(self, *, step_ctx, iteration_results=None):
             results[NUM_AGENT_STEPS_TRAINED + "_this_iter"] = step_ctx.trained
             # TODO: For CQL and other algos, count by trained steps.
             results["timesteps_total"] = self._counters[NUM_AGENT_STEPS_SAMPLED]
+            # TODO: Backward compatibility.
+            results[STEPS_TRAINED_THIS_ITER_COUNTER] = step_ctx.trained
```

Comment:
Make sure old keys are still available in the metrics, but do NOT require user to keep track of these in the `training_step()` methods.